dotCMS/src/main/java/com/dotmarketing/db/listeners/CommitAPI.java

```diff
@@ -0,0 +1,349 @@
+        syncCommitListeners.get().remove(listener.key(), listener);
+        syncCommitListeners.get().put(listener.key(), listener);
+    }
+
+    /**
+     * adds this to the end of sync CommitListener list
+     */
+    public void addReindexListenerSync(final ReindexListener listener) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
+        syncCommitListeners.get().remove(listener.key(), listener);
+        syncCommitListeners.get().put(listener.key(), listener);
+    }
+
+    /**
+     * adds this to the end of async CommitListener list
+     */
+    public void addReindexListenerAsync(final ReindexListener listener) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
+        asyncCommitListeners.get().remove(listener.key(), listener);
+        asyncCommitListeners.get().put(listener.key(), listener);
+    }
+
+    /**
+     * adds this to the end of sync CommitListener list
+     */
+    public void addFlushCacheSync(final FlushCacheListener listener) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
+        syncCommitListeners.get().remove(listener.key());
+        syncCommitListeners.get().put(listener.key(), listener);
+        addRollBackListener(listener);
+    }
+
+    /**
+     * adds this to the end of async CommitListener list
+     */
+    public void addFlushCacheAsync(final FlushCacheListener listener) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
+        asyncCommitListeners.get().remove(listener.key());
+        asyncCommitListeners.get().put(listener.key(), listener);
+        addRollBackListener(listener);
+
+    }
+
+    /**
+     * adds this to the end of rollback CommitListener list
+     */
+    public void addRollBackListener(final RollbackListener runable) {
+        if (!DbConnectionFactory.inTransaction())
+            return;
+        rollbackListeners.get().remove(runable.key(), runable);
+        rollbackListeners.get().put(runable.key(), runable);
+    }
+
+    /**
+     * adds this to the end of rollback CommitListener list
+     */
+    public void addRollBackListener(final DotListener listener) {
+        addRollBackListener(new RollbackListener(listener));
+    }
+
+    /**
+     * clears the listeners and resets the state to ALLOW_ASYNC
+     */
+    private void resetListeners() {
+        rollbackListeners.get().clear();
+        syncCommitListeners.get().clear();
+        asyncCommitListeners.get().clear();
+        setCommitListenerStatus(nextStatus);
+        nextStatus = CommitListenerStatus.ALLOW_ASYNC;
+    }
+
+
+    /**
+     * This method is fired AFTER a commit and BEFORE a connection is closed and returned to the pool
+     */
+    public void finalizeListeners() {
+        try {
+            if (DbConnectionFactory.inTransaction()) {
+                throw new DotStateException("Commit Listeners need run after a commit has taken place");
+            }
+            switch (status()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5639225)